### ISSUE TITLE
Python multithreading print error with the integrated terminal 

### ISSUE THREAD

**Issue by Meano** [Microsoft/vscode#40199]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:32:36.023Z)
- OS Version: Windows_NT x64 10.0.16299
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
code-gnu-global|aus|0.2.2
python|ms-|0.8.0
cpptools|ms-|0.14.4

---

Steps to Reproduce:

1. create python script:
```
from multiprocessing import Pool, cpu_count
import traceback
import sys

def printfunc(msg):
    for i in range(20):
        print "TEST "+ str(msg) +" Count:" + str(i)
    return 0

if __name__ == "__main__":
    try:
        testlist = list(range(1000))
        pool_size = cpu_count()
        pool = Pool(processes=pool_size)
        print "Pool Size:", pool_size
        pool.map(printfunc, testlist)
        pool.close()
        pool.join()
    except Exception,e:
        pool.terminate()
        traceback.print_exc()
        sys.exit(1)
```
2. execute the script  in integrated terminal will print error message like:

![tim 20171214160626](https://user-images.githubusercontent.com/7509521/33981680-c3cfb20c-e0e8-11e7-89fb-3e0cd14505f5.png)

3. but the script has no errors execute in external terminal (cmd or powershell)

Reproduces without extensions: **Yes**

**Comment by Tyriar** [comment]
@ivankravets looks like this is the same as https://github.com/platformio/platformio-vscode-ide/issues/61?

**Comment by ivankravets** [comment]
@Tyriar Yes, the same. In our case, the problem is linked with `urllib2` library. However, the root of a problem is the same `IOError: [Errno 0] Error`.

This is a very specific issue. I can't reproduce it on my Windows VMs :(

**Comment by Meano** [comment]
@ivankravets I've tested it in windows7 and the issue can't be reproduced. It seems to be related to the OS version. And the problem may happened after I update to Windows 10 1703.

**Comment by jackietw** [comment]
Linux version and MacOS version seem no this issue, Windows version, I just only running on Windows 10, other Windows I haven't.

**Comment by ivankravets** [comment]
@Meano does Python's `socket` work for you? Could you try this script
```python
from urllib2 import urlopen

url = "http://dl.platformio.org/packages/toolchain-xtensa-windows-1.40802.0.tar.gz"

r = urlopen(url)
i = 0
for chunk in r.read(1024):
    i += 1
    print i
```

It fails for me between 10-30 iterations.

**Comment by Meano** [comment]
![tim 20171215194955](https://user-images.githubusercontent.com/7509521/34041048-60affd14-e1d1-11e7-8e40-1b3ec03f3427.png)

Failed, print error.

**Comment by ivankravets** [comment]
Cool! Let's wait when @Tyriar will fix it. I don't know why Python users don't report this issue. Maybe, they don't use built-in terminal :)

**Comment by Tyriar** [comment]
@rprichard any idea what could be happening here? Interestingly it doesn't seen to happen with @ivankravets's [pty.js fork](https://github.com/platformio/pty.js) which is still on winpty@0.2.

**Comment by kzhui125** [comment]
```
for num in range(0,1000):
    print str(num)
```
using python 2.7, integrated terminal is unusable now

**Comment by rprichard** [comment]
@Tyriar: @segevfiner filed an upstream winpty issue (https://github.com/rprichard/winpty/issues/134). I looked into it, and it looks like a Windows bug, which I filed here: https://github.com/Microsoft/console/issues/40. I doubt winpty can work around this problem.

FWIW, the legacy console shouldn't be affected by this issue.

It looks like Python 3 is also unaffected. Maybe it uses `WriteConsoleW` instead of `WriteFile`.
